mentat/llm_api.py

```diff
@@ -104,11 +110,31 @@ def warn_user(message: str, max_tries: int, details: Details):
     on_backoff=partial(warn_user, "Rate limit recieved from OpenAI's servers", 3),
 )
 async def call_llm_api(
-    messages: list[dict[str, str]], model: str
+    messages: list[dict[str, str]], model: str | None = None
 ) -> AsyncGenerator[Any, None]:
     raise_if_in_test_environment()
     session_context = SESSION_CONTEXT.get()
+    stream = session_context.stream
     config = session_context.config
+    if model is None:
+        model = config.model
+    if not is_model_available(model):
+        raise MentatError(
+            f"Model {model} is not available. Please try again with a different model."
+        )
+    if "gpt-4" not in model:
```

Comment:
I think the main reason we sent this message at the start was because we only wanted to print it once. Maybe we should print this warning any time config.model is changed (including at the start)? This might be another reason to have a specific set_config function in the config (instead of using setattr)